packages/react/src/theming/extendTheme.tsx

```diff
@@ -0,0 +1,9 @@
+import merge from 'deepmerge';
+
+import { getCSSVariablesFromTheme } from './utils';
+
+export const extendTheme = <T, K>(themeObject: T, override: K) => {
```

Comment:
1. I was thinking the `themeObject` doesn't necessarily to be our `Theme`, so the user can bring in their own theme. But I don't know the long term goal here. Thoughts? @dbanksdesign 

2. Maybe the name `override` here is a bit confusing but essentially the `override` could just be something like `{ colors: brand: { tertiary: #xxxx } }`. It could be entirely a custom object(either partial `Theme` or something we don't define before) and doesn't necessarily to be a `Theme`. Any thoughts for the naming?

Reply:
Really all we are doing is providing a deepMerge functionality, merge overrides into themeObject, right? I guess maybe more than that (out of scope for this PR) we could provide reference resolution logic, similar to what Style Dictionary does so that the themeObject and overrides could contain references that get resolved in the `extendTheme` method... just thinking out loud.. 

1. Customers could publish a default Amplify theme other customers could use instead of our default theme. We (Amplify) could even publish multiple default themes customers could use.
2. Provide a way to extend a theme by overriding certain parts of it, while keeping references intact. For example `components.badge.info.backgroundColor` -> `colors.background.info` -> `colors.blue.20`, a user should be able just give us `colors.background.info` or `colors.blue.20` and the info badge should be updated. 

I could also see us supporting light/dark mode at the theme level by swapping themeObjects in the provider.

Reply:
1. If a user has their own `theme`, then what is the use case for overriding values of their custom theme?

2. IMO `overrideTheme: Theme` should suffice. Regarding extending the theme to include keys that are not a part of the base theme, it seems like that is a separate concern. Thoughts?

Reply:
> > Edit: these replies are to @zchenwei
> 
> 1. If a user has their own `theme`, then what is the use case for overriding values of their custom theme?
> 2. IMO `overrideTheme: Theme` should suffice. Regarding extending the theme to include keys that are not a part of the base theme, it seems like that is a separate concern. Thoughts?

I am good to type it as Theme. I think we should go easy at the beginning :)

Reply:
> Really all we are doing is providing a deepMerge functionality, merge overrides into themeObject, right? I guess maybe more than that (out of scope for this PR) we could provide reference resolution logic, similar to what Style Dictionary does so that the themeObject and overrides could contain references that get resolved in the `extendTheme` method... just thinking out loud..
> 
> 1. Customers could publish a default Amplify theme other customers could use instead of our default theme. We (Amplify) could even publish multiple default themes customers could use.
> 2. Provide a way to extend a theme by overriding certain parts of it, while keeping references intact. For example `components.badge.info.backgroundColor` -> `colors.background.info` -> `colors.blue.20`, a user should be able just give us `colors.background.info` or `colors.blue.20` and the info badge should be updated.
> 
> I could also see us supporting light/dark mode at the theme level by swapping themeObjects in the provider.

I like the ideas. We should introduce the reference resolution logic within this function in the near future